protocol/xmrtaker/net.go

```diff
@@ -17,47 +17,49 @@ func (a *Instance) Provides() types.ProvidesCoin {
 // The input units are ether that we will provide.
 func (a *Instance) InitiateProtocol(providesAmount float64, offer *types.Offer) (common.SwapState, error) {
 	receivedAmount := offer.ExchangeRate.ToXMR(providesAmount)
-	err := a.initiate(common.EtherToWei(providesAmount), common.MoneroToPiconero(receivedAmount),
+	state, err := a.initiate(common.EtherToWei(providesAmount), common.MoneroToPiconero(receivedAmount),
 		offer.ExchangeRate, offer.GetID())
 	if err != nil {
 		return nil, err
 	}
 
-	return a.swapStates[offer.GetID()], nil
```

Comment:
This access to `a.swapStates` should have been grabbing the lock.  Issue was fixed by having `a.initiate` return the swap state (analogous change was made in `xmrmaker` earlier).